File: frontend/src/types/sprint.ts
export type SprintStatus = "backlog" | "active" | "closed"

export interface Sprint {
  id: string
  organization_id: string
  project_id: string
  name: string
  description: string | null
  status: SprintStatus
  start_date: string | null
  end_date: string | null
  goals: string | null
  task_count: number
  done_count: number
  total_points: number
  created_at: string
  updated_at: string
}

export const SPRINT_STATUS_LABELS: Record<SprintStatus, string> = {
  backlog: "Backlog",
  active: "Active",
  closed: "Closed",
}

File: app/schemas/sprint.py
"""Pydantic schemas for sprint endpoints."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.sprint import SprintStatus


class SprintCreateRequest(BaseModel):
    name: str = Field(..., min_length=1, max_length=200)
    description: str | None = Field(None, max_length=2000)
    start_date: datetime | None = None
    end_date: datetime | None = None
    goals: str | None = Field(None, max_length=2000)


class SprintUpdateRequest(BaseModel):
    name: str | None = Field(None, min_length=1, max_length=200)
    description: str | None = None
    status: SprintStatus | None = None
    start_date: datetime | None = None
    end_date: datetime | None = None
    goals: str | None = None


class SprintOut(BaseModel):
    id: uuid.UUID
    organization_id: uuid.UUID
    project_id: uuid.UUID
    name: str
    description: str | None
    status: SprintStatus
    start_date: datetime | None
    end_date: datetime | None
    goals: str | None
    task_count: int = 0
    done_count: int = 0
    total_points: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
